lib/helpers.rb

```diff
@@ -65,15 +65,15 @@ def un_flag_as_abuse(obj)
     raise ArgumentError, t(:user_id_is_required) unless user
     if params["all"]
       # If historical abuse flags didn't exist so far, increment them.
-      obj.author.update_stats_for_course(obj.course_id, inactive_flags: 1) unless obj.historical_abuse_flaggers
+      obj.author.update_stats_for_course(obj.course_id, inactive_flags: 1) unless obj.historical_abuse_flaggers.empty?
       obj.historical_abuse_flaggers += obj.abuse_flaggers
       obj.historical_abuse_flaggers = obj.historical_abuse_flaggers.uniq
       obj.abuse_flaggers.clear
     else
       obj.abuse_flaggers.delete user.id
     end
     # If this operation results in no more abuse flags, decrement the cour.
-    obj.author.update_stats_for_course(obj.course_id, active_flags: -1) unless obj.abuse_flaggers
+    obj.author.update_stats_for_course(obj.course_id, active_flags: -1) unless obj.abuse_flaggers.empty?
```

Comment:
This should be `if obj.abuse_flaggers.empty?` instead of `unless`, if I understand correctly.